src/content.rs

```diff
@@ -27,9 +28,9 @@ impl Content {
     }
 
     /// Return the raw constructed byte stream.
-    pub fn finish(mut self) -> Vec<u8> {
```

Comment:
It would be good to add a little documentation here what one is supposed to do with the `Buf`.

In particular, the examples in `examples/` also currently discard the limits information when writing content streams. It'd be nice if we could make that more type-safe. E.g. by making it an explicit opt-in operation to write a stream without limits (ideally though without forcing an allocation if you want to write an `&[u8]`).

Reply:
> It'd be nice if we could make that more type-safe. E.g. by making it an explicit opt-in operation to write a stream without limits (ideally though without forcing an allocation if you want to write an &[u8]).

I wanted to do this, but it is kind of hard, because as soon as you need to apply a filter to the stream (which you need in most cases) you have to operate on the raw bytes anyway. But definitely will still add some more documentation (and a test)

Reply:
Hmm.. So I've added some documentation (not pushed yet), but do you really think it's a good idea to _force_ the user to provide this each time they write a stream? I feel like this would pollute the API quite a bit. Keeping track of the limits is a really niche feature, so I feel like it's okay if it's not that prominent in the API. Maybe it would be enough to just add a new example?

Reply:
Yeah okay